test/extensions/key_value/file_based/key_value_store_test.cc

```diff
@@ -46,7 +51,20 @@ TEST_F(KeyValueStoreTest, Basic) {
   store_->remove("foo");
   EXPECT_EQ(absl::nullopt, store_->get("foo"));
 }
+TEST_F(KeyValueStoreTest, TTL) {
+  EXPECT_EQ(absl::nullopt, store_->get("foo"));  
 
+  store_->addOrUpdate("foo", "bar", std::chrono::milliseconds(5));
+  EXPECT_EQ("bar", store_->get("foo").value());
+  // Advance timer, but not enough to trigger expiry
+  test_time_.setSystemTime(std::chrono::milliseconds(1));
+  ttl_timer_->invokeCallback();
```

Comment:
In production, would we expect the timer to fire at this point? I would have expected that it would be armed to fire after 5ms at which point it would do the eviction. Perhaps we can add an EXPECT_CALL like this:
`EXPECT_CALL(*ttl_timer_, enableTimer(std::chrono::milliseconds(timeout), _));`
just before we call `addOrUpdate()`.

Reply:
Maybe I missed this but did you add an EXPECT_CALL?